pytorch_lightning/utilities/enums.py

```diff
@@ -188,3 +231,41 @@ def get_max_depth(mode: str) -> int:
     @staticmethod
     def supported_types() -> List[str]:
         return [x.value for x in ModelSummaryMode]
+
+
+class _StrategyType(LightningEnum):
+    """Define type of training strategy.
+
+    >>> # you can match the type with string
+    >>> _StrategyType.DDP == 'ddp'
+    True
+    >>> # which is case invariant
+    >>> _StrategyType.DDP2 in ('ddp2', )
+    True
+    """
+
+    DP = "dp"
+    DDP = "ddp"
+    DDP2 = "ddp2"
+    DDP_CPU = "ddp_cpu"
+    DDP_SPAWN = "ddp_spawn"
+    TPU_SPAWN = "tpu_spawn"
+    DEEPSPEED = "deepspeed"
+    HOROVOD = "horovod"
+    DDP_SHARDED = "ddp_sharded"
+    DDP_SHARDED_SPAWN = "ddp_sharded_spawn"
+    DDP_FULLY_SHARDED = "ddp_fully_sharded"
+
+    @staticmethod
+    def interactive_compatible_types() -> List["_StrategyType"]:
+        """Returns a list containing interactive compatible _StrategyTypes."""
+        return [
+            _StrategyType.DP,
+            _StrategyType.DDP_SPAWN,
+            _StrategyType.DDP_SHARDED_SPAWN,
+            _StrategyType.TPU_SPAWN,
+        ]
+
+    def is_interactive_compatible(self) -> bool:
```

Comment:
By `interactive`, do you mean notebooks ? Won't `is_notebook_compatible` more explicit ?

Reply:
Imo, it's an industry standard of it being called `interactive`